src/front/wgsl/parse/mod.rs

```diff
@@ -142,6 +143,11 @@ impl BindingParser {
                 self.location = Some(Parser::non_negative_i32_literal(lexer)?);
                 lexer.expect(Token::Paren(')'))?;
             }
+            "index" => {
+                lexer.expect(Token::Paren('('))?;
+                self.index = Some(Parser::non_negative_i32_literal(lexer)?);
+                lexer.expect(Token::Paren(')'))?;
+            }
```

Comment:
- would you just have an attribute called bind_src_1? like this?
```
struct FragmentOutput{
    @location(0) color: vec4<f32>,
    @location(0) @bind_src_1 mask: vec4<f32>,
};
```
Ideally it would be nice to keep to whatever the wgpu spec https://www.w3.org/TR/WGSL/ has standardized but i don't know how the procedure to get some changes there is. It is probably too cumbersome, so it is better to just stay proprietary for now.

Reply:
> * would you just have an attribute called bind_src_1? like this?
> 
> ```
> struct FragmentOutput{
>     @location(0) color: vec4<f32>,
>     @location(0) @bind_src_1 mask: vec4<f32>,
> };
> ```

Yes, but `blend_src_1` not bind.

> Ideally it would be nice to keep to whatever the wgpu spec https://www.w3.org/TR/WGSL/ has standardized but i don't know how the procedure to get some changes there is. It is probably too cumbersome, so it is better to just stay proprietary for now.

I will advocate for `blend_src_1` but if it gets standardized as something else we can change our implementation.

> Would you like to translate @bind_src_1 to index 1 internally, or would you remove the index parameter on location in the generic data structure and rather have a bind_src_1 bool?

Yeah, I think that makes the most sense.

Reply:
I used alternate_blend_source for the internal variable, as i think that reflects better what it is, although if you insist i can do a rename of the variables. I used blend_src_1 for the wgsl tough as it reflects the SRC1 when setting up the blend function.